metric_config_parser/config.py

```diff
@@ -496,13 +496,19 @@ def as_of(self, timestamp: datetime) -> "ConfigCollection":
                 # make sure the most recent commit is checked out by checking out the main branch
                 tmp_repo.git.checkout(repo.main_branch)
 
+                # keep track of more recent commits to go back to in case invalid configs
+                # got checked into main that cannot be parsed
+                newer_commits: List[str] = []
+
                 # start iterating through all commits starting at HEAD
                 for commit in tmp_repo.iter_commits("HEAD"):
                     # check if the commit timestamp is older than the reference timestamp
                     if commit.committed_datetime <= timestamp:
                         rev = commit.hexsha
                         break
 
+                    newer_commits.insert(0, commit.hexsha)
```

Comment:
If you put the desired commit in here too, you could simplify the later logic to just start with the loop part.